### PR TITLE
Update cube size based on caliper measurements.

### PR DESCRIPTION
**Pull request by IamfromSpace**

### CODE REVIEW

**Comment by IamfromSpace** [comment]
Only question is do we want to include tolerance in the measurement or do we want it to be exact and include tolerance as needed?

**Comment by MichaelRFairhurst** [comment]
Probably want to include tolerance as needed? I'm thinking, for instance, that tolerance is a property of the printer and the specific shape in question and the play in the structure. So our cube_holder() may need 1mm of tolerance, but our static claw may need 3mm of tolerance. So it seems like we shouldn't "double" them.

That said...I'm not positive it isn't a good idea to have some basic tolerance. Basically, from the perspective of, "this printer is designed for a cube up to 57mm wide...however, most cubes are a bit smaller, which is fine" type of thing.

**Review by MichaelRFairhurst — APPROVED** [review]
approved, however you want to do it, I gave you my thoughts and have no clear reason to say they aren't dumb

**Comment by IamfromSpace** [comment]
Yeah, tolerance is weird because it’s really a manufacturing detail—which inevitably affects the design.

Probably makes more sense to pull it out, and have it be a module input—similar to the way $fn works.  Maybe we can make it a constant, since it should be about the extrusion size for 3D printing but would be much lower for something that was CNC’d.

**Comment by MichaelRFairhurst** [comment]
Yeah I have definitely learned though that tolerances, at least in 3d printing, are very strange. Like, they can cumulative in different ways, they can depend upon different things like the physical results of the printer (which in itself is complicated: globs, warping, overhang, bridging, overextrusion, striations), material flex, temperature, postproduction like sanding...

Using a $ var like $fn (you can make your own) is an interesting idea! I suppose you could even go so far as to like, multiply them or regress them etc in places where it made sense to do so.

I wonder how professional engineers do it...

**Comment by IamfromSpace** [comment]
Updated!  Let me know what you think.  I'd also update #18 once merged to use `$tol` everywhere that `tolerance` is used now--which I think is much more consistent/manageable.

File: torque_tester.scad [hunk 1]
```diff
@@ -22,10 +22,11 @@ torque_tester(_lever_length, $fn=30);
 // weight/lever_length to move the cube.
 module torque_tester(lever_length, cube_size=cube_size, layer_count=3, wall_thickness=2) {
   depth = cube_size / (layer_count * 1.33);
-  width = cube_size + wall_thickness * 2;
+  tolerant_cube_size = cube_size + $tol;
+  width = tolerant_cube_size + wall_thickness * 2;
   difference() {
     cube([width, depth, width], center=true);
-    cube([cube_size, 3*depth, cube_size], center=true);
+    cube([tolerant_cube_size, 3*depth, tolerant_cube_size], center=true);
```
Comment (MichaelRFairhurst) [inline torque_tester.scad:29]: how about a function which takes a value and/or vector?

```
// in util.scad:
function tol(x) = isnum(x) ? x + $tol : vtol(x);
function vtol(x) = [ for i [0:len(x) -1] tol(i) ];
...
// here:
  cube(tol([cube_size, 3*depth, cube_size]), center=true);
// or even just:
  cube(tol(cube_size), center=true);
```

**Review by MichaelRFairhurst — APPROVED** [review]
Still approved, still more rambling thoughts!

File: torque_tester.scad (hunk 1 above)
Comment (IamfromSpace) [reply torque_tester.scad:29]: Interesting, makes sense!  I'm going to merge these as is, but let's definitely add to utils to help manage tolerances!

**Pull request merged**